feat(button): enhance button documentation with more user guidelines … (#62)
commercetools/nimbus: packages/nimbus/src/components/button/button.mdx

# Button

---

export const Tip = ({ icon, tipColor, tipBackgroundColor, children }) => (
  <div
    style={{
      backgroundColor: tipBackgroundColor,
      borderLeft: `4px solid ${tipColor}`,
      color: tipColor,
      padding: "20px 13px",
      display: "flex",
      gap: "11px",
      mergin: 0,
    }}
  >
    <span style={{ color: "inherit", fontSize: "24px" }}>{icon}</span>
    {children}
  </div>

> );

Author: Thank you! As agreed on our DS forum, we may not use this for now.

> );

Reviewer: I know we talked about not using these, but I think it's a superior solution to the custom `Tip` component that would have to live in every file (or extracted out to use everywhere).

```
> [!TIP]\
> **Do**
>
> - Use the correct and consistently patterned icon for the action that a user
>   will be selecting.
> - Ok ways to break this rule are with drop downs and right pointing arrows.
```

After spending an hour and half pairing with Jai, we came to the conclusion that this will only with either two spaces after the closing bracket or a backslash. The backslash is better because it isn't hidden.

> );

Author: Yep! this works well now. Thank you for giving this a look! 👀

A button serves as a standardized, reusable component for triggering actions and
navigating.

## Overview

Buttons are clickable elements that initiate actions. They're how we encourage
users to take action and explore different parts of our interface.

### Resources

Deep dive on details and access design library

[Figma library](https://example.com) <a href="https://example.com"
style={{color: "blue"}}>Figma library</a>

### Hierachy

Deep dive on details and access design library

#### Primary actions

Primary tone, filled style, medium size. For the principal call to action on the
page. Primary buttons should only appear once per screen.

> - **A solid button** should be used as a primary action on a page used to

Author: Thank you for catching this. wrt to our DS forum discussions, we do not really need to get the styling of the content exactly as it is in the design.

  visually prominent than a primary or secondary button. These can apply to
  tertiary actions such as drop downs and contextual actions.
- **Link buttons** can be used when the focus should be on the content rather
  than the UI controls themselves. They are less visually distracting and allow
  the content to take center stage. This is common in content-heavy applications
  or dashboards. Link and Ghost styled buttons are most used in tertiary
  actions, and are styled in X-small or 2X-small sizes.

```jsx-live
const App = () => (
  <Stack direction="horizontal">
    <Button tone="primary" size="md" variant="ghost">
      Filter
    </Button>
    <Button tone="primary" size="xs" variant="link">
      Expand options
      <Icons.KeyboardArrowDown />
   </Button>
 </Stack>
)
```

### Variables

Get familiar with the features.

### Icons in buttons

**Icons before or after in buttons**

The default position of icons in primary and secondary buttons are to the left
of the text, but we allow for icons to follow for special uses such as drop
downs, steppers, and expansion links. For icon placement best practices refer to

> our guidelines.

Author: Very good point, I think we would need to document this somewhere so we get back to it. I have documented this.

  average 20% more length.

### Preferred words in buttons

| **Text**                                | **Description**                                                                                                                                                                                                                                                                                                                          |
| --------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| **Save**                                | Carries out pending changes, commits to saving the information.                                                                                                                                                                                                                                                                          |

> | **Back**                                | This term typically implies a return to a previous state or location in a non-linear or less structured flow. It's often used for:<ul style={{listStyle: "disc"}}><li>Navigating browser history</li><li>Exiting a screen or section</li></ul>                                                                                           |

Author: Yeah, I think this would require some extra unnecessary styling inside of the table to achieve. Per our DS forum discussion, I think this is fine for now. 

Id have to do something like this for every cell in the column 
| **Text**                    | **Use this instead**                                                                                   |
|----------------------------|--------------------------------------------------------------------------------------------------------|
| **New**                    | <span style={{ color: "#0070f3" }}>Be more specific about what the user is creating.</span>             |
| **Continue**               | <span style={{ color: "#0070f3" }}>Use “Next” instead of “Continue” instead to maintain consistency.</span> |
| **Create new (noun)**      | <span style={{ color: "#0070f3" }}>Use “Add (noun)” instead of “create” in button language to maintain consistency.</span> |
| **OK, Sure, Fine, Yes, No**| <span style={{ color: "#0070f3" }}>Use words that explain the action.</span>                            |
| **Wow! Yes? DELETE NOW!**  | <span style={{ color: "#0070f3" }}>Avoid full capitalization and punctuation.</span>                   |

        </div>
        </Card.Content>
      </Card.Root>
    </Stack>
  </Stack>
);
```

### Button groupings

Button groups are UI elements that organize multiple buttons together, often
visually as a single unit. They serve to present users with a set of related
choices or actions in a compact and organized way.

<Tip tipColor="#46A557" tipBackgroundColor="#F4FAF5" icon={<Icons.Lightbulb />}>
  <div>
    <strong>Do</strong>
    <ul style={{ listStyle: "disc", marginLeft: "20px" }}>
      <li>
        This is the correct hierarchy of button groupings. From right to left,
        Primary, Outlined, and Link button.
      </li>
    </ul>
  </div>
</Tip>

```jsx-live
  const App = () => (
  <Stack direction="horizontal">
    <Button tone="primary" as="a" size="md" variant="link">
      Calcel and close
    </Button>
    <Button tone="primary" size="md" variant="outline">
      <Icons.Undo />
      Revert
    </Button>
    <Button tone="primary" size="md" variant="solid">
      Save changes
    </Button>
  </Stack>
)
```

Button groups are UI elements that organize multiple buttons together, often
visually as a single unit. They serve to present users with a set of related

> choices or actions in a compact and organized way.

Author: Good catch!

alternative action compared to the primary button.

- **An outlined button** should be used as for secondary actions that are less
  important or less frequent than the primary action. They provide users with
  alternative options without distracting from the main call to action.

```jsx-live
const App = () => (
  <Button tone="primary" colorPalette="primary" variant="outline" size="md">
   <Icons.Replay />
   Secondary action
 </Button>
)
```

#### Tertiary actions

Primary tone, link styled, xsmall size. For lesser-performed actions or a
collection of actions — those that users are unlikely to perform.

- **Ghost buttons** are used when you want a button to be visible but less
  visually prominent than a primary or secondary button. These can apply to
  tertiary actions such as drop downs and contextual actions.
- **Link buttons** can be used when the focus should be on the content rather
  than the UI controls themselves. They are less visually distracting and allow
  the content to take center stage. This is common in content-heavy applications
  or dashboards. Link and Ghost styled buttons are most used in tertiary
  actions, and are styled in X-small or 2X-small sizes.

> ```jsx-live

Reviewer: The bottom margin on these is silly large, is there any way to reduce it?

Author: I think that is the default we can get at the moment. Theres really no styling anywhere.

const App = () => (

>   <Stack direction="horizontal">

Reviewer: ```suggestion
  <Stack direction="horizontal" alignItems="center">
```

Nit nit nit

      <Icons.SentimentSatisfied />
    </Button>
    <Button tone="primary" size="md" variant="outline">
      <Icons.SentimentSatisfied />
      Outline
      <Icons.SentimentSatisfied />
   </Button>
    <Button tone="primary" size="md" variant="ghost">
      <Icons.SentimentSatisfied />
      Ghost
      <Icons.SentimentSatisfied />
   </Button>
    <Button tone="primary" size="md" variant="link">
      <Icons.SentimentSatisfied />
      Link
      <Icons.SentimentSatisfied />
   </Button>
 </Stack>
)
```

### Size

**Medium** (40px)

This is the default and preferred size for buttons.

```jsx-live
const App = () => (
  <Button colorPalette="primary" size="md" variant="solid">
   Save
 </Button>
)
```

**X-Small** (32px)

If there is a space constraint, use these buttons. Ideally will be used
sparingly.

```jsx-live
const App = () => (
 <Button colorPalette="primary" size="xs" variant="solid">
   Apply Filters
 </Button>
)
```

**2x-Small** (24px)

Used mainly for Ghost and Link buttons, these buttons can be seen in filters
where the size constraint is tightest.

```jsx-live
const App = () => (
  <Button colorPalette="primary" size="2xs" variant="ghost">
   <Icons.Edit />

>    Primary action

Reviewer: ```suggestion
   Edit filters
```

- Be aware that with localization/translation character count will add on
  average 20% more length.

### Preferred words

| **Text**                                | **Description**                                                                                                                                                                                                                                                                                              |
| --------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
| **Save**                                | Carries out pending changes, commits to saving the information.                                                                                                                                                                                                                                              |

> | **Back**                                | This term typically implies a return to a previous state or location in a non-linear or less structured flow. It's often used for:<ul style={{listStyle: "disc"}}><li>Navigating browser history</li><li>Exiting a screen or section</li></ul>                                                               |

Reviewer: ```suggestion
| **Back**                                | This term typically implies a return to a previous state or location in a non-linear or less structured flow. It's often used for:<ul style={{listStyle: "disc", marginLeft: "20px"}}><li>Navigating browser history</li><li>Exiting a screen or section</li></ul>                                                               |
```

Author: Such a simple trick, why didn't I think of that! Thank you.

> | **Previous**                            | This term is more common in linear, sequential flows, such as: <ul style={{listStyle: "disc"}}><li>Stepping through a list of items</li><li>Moving between pages in a paginated view</li><li>Progressing through a multi-step process</li></ul> <br />Pair “Previous” with “Next” button language.           |

Reviewer: ```suggestion
| **Previous**                            | This term is more common in linear, sequential flows, such as: <ul style={{listStyle: "disc", marginLeft: "20px"}}><li>Stepping through a list of items</li><li>Moving between pages in a paginated view</li><li>Progressing through a multi-step process</li></ul> <br />Pair “Previous” with “Next” button language.           |
```

> | **Add new (noun)**                      | <ul><li>Establishes a new object from scratch. Always followed by an object. For example “add product” “add category”.</li><li>Can also be used in areas that establishes a new relationship. Example “add products to a category”.</li><li>Do not use “add new” as it does not give full context.</li></ul> |

Reviewer: ```suggestion
| **Add new (noun)**                      | Establishes a new object from scratch. Always followed by an object. For example “add product”, “add category”.<br/><br/>Can also be used in areas that establishes a new relationship. Example “add products to a category”.<br/><br/>Do not use “add new” as it does not give full context. |
```

| **Continue**                  | Use “Next” instead of “Continue” instead to maintain consistency.                |
| **Create new (noun)**         | Use “Add (noun)” instead of “create” in button language to maintain consistency. |
| **OK, Sure, Fine, Yes, No**   | Use words that explain the action.                                               |
| **Wow! Yes? DELETE NOW! No?** | Avoid full capitalization and punctuation.                                       |

### Usage

**A primary filled button** highlights the most important action a user can
take. For example, it might be used to save changes, submit a form, or emphasize
a key action like adding a discount.

**Secondary and tertiary** (primary toned outline and linked styling) buttons
are used when there's already a primary action on the page. They're for less
prominent actions, like canceling changes, supporting the primary action, going
back in a process, or expanding/collapsing information.

**Critically toned buttons** are reserved for destructive actions, such as
deleting something or removing a connection.

<Tip tipColor="#46A557" tipBackgroundColor="#F4FAF5" icon={<Icons.Lightbulb />}>
  <>When to use</>

> </Tip>
> <br />

Reviewer: ```suggestion
</Tip>
```

I think this looks nicer

- **Trigger an action:** Anything that happens (save, delete, submit, download,
  etc.).
- **Control a process:** Start, stop, pause, resume, undo, redo.
- **Confirm a choice:** "Yes," "Save changes," Confirm.
- **Open something:** Modal, popover, dialog.
- **Select an option:** Especially complex or visual choices.

>   <br />

Reviewer: ```suggestion
<br />
```

<Tip tipColor="#E5484D" tipBackgroundColor="#FFF5F5" icon={<Icons.Error />}>
  <>When not to use</>

> </Tip>
> <br />

Reviewer: ```suggestion
</Tip>
```

    </Button>
    <Button tone="critical" size="md" variant="solid">
      <Icons.Clear />
        Add discount
      <Icons.Clear />
    </Button>
    <Button tone="primary" size="md" variant="solid">
      Revert
      <Icons.Undo />
    </Button>
 </Stack>
)
```

<Tip tipColor="#46A557" tipBackgroundColor="#F4FAF5" icon={<Icons.Lightbulb />}>
  <div>
    <strong>Do</strong>
    <ul style={{ listStyle: "disc", marginLeft: "20px" }}>
      <li>
        Use right icons to build a drop down component with the Popover
        component.
      </li>
      <li>
        Icons are on the left unless there is a directional reason to use a
        right icon placement.
      </li>
    </ul>
  </div>
</Tip>
<br />

```jsx-live
const App = () => (
  <Stack direction="horizontal">
    <Stack>
      <Button tone="primary" size="md" variant="outline">
        Save changes
        <Icons.KeyboardArrowDown />
      </Button>
     <Card.Root elevation="elevated" cardPadding="md">
        <Card.Content>
        <div style={{fontSize: "16px", fontWeight: "400"}}>
        <p>Save</p>
        <p>Save and publish</p>
        <p>Save and open next</p>
        </div>
        </Card.Content>

>       </Card.Root>

Reviewer: This isn't actually how we're doing buttons with dropdowns, right? 😅

Author: Nope. Otherwise, we would just use the select in the snippet instead. This is mainly for the visuals and we may need to look into how to turn off the snippets for cases like this.

Reviewer: This is perhaps a stupid question. Is this documentation appropriate for this location? Are you saying that someone could never have a dropdown open from a button? This particular question does not need to hold up this PR.

      </Card.Root>
    </Stack>
    <Stack>
      <Button tone="primary" size="md" variant="solid">
        <Icons.KeyboardArrowDown />
        Export categories
        <Icons.FileUpload />
      </Button>
      <Card.Root elevation="elevated" cardPadding="md">
        <Card.Content>
        <div style={{fontSize: "16px", fontWeight: "400"}}>
        <p>Export as csv</p>
        <p>Download to drive</p>
        </div>
        </Card.Content>
      </Card.Root>
    </Stack>
  </Stack>
);
```

### Button groupings

Button groups are UI elements that organize multiple buttons together, often
visually as a single unit. They serve to present users with a set of related
choices or actions in a compact and organized way.

<Tip tipColor="#46A557" tipBackgroundColor="#F4FAF5" icon={<Icons.Lightbulb />}>
  <div>
    <strong>Do</strong>
    <ul style={{ listStyle: "disc", marginLeft: "20px" }}>
      <li>
        This is the correct hierarchy of button groupings. From right to left,
        Primary, Outlined, and Link button.
      </li>
    </ul>
  </div>
</Tip>

```jsx-live
  const App = () => (
  <Stack direction="horizontal">
    <Button tone="primary" as="a" size="md" variant="link">

>       Calcel and close

Reviewer: ```suggestion
      Cancel and close
```

    </Button>
    <Button tone="primary" size="md" variant="outline">
      <Icons.Undo />
      Revert
    </Button>
    <Button tone="primary" size="md" variant="solid">
      Save changes
    </Button>
  </Stack>
)
```

<Tip tipColor="#E5484D" tipBackgroundColor="#FFF5F5" icon={<Icons.Error />}>
  <div>
    <strong>Don't</strong>
    <ul style={{ listStyle: "disc", marginLeft: "20px" }}>
      <li> Avoid stacking actions.</li>
      <li>Don’t use two filled primary toned buttons in the same area. </li>
    </ul>
  </div>
</Tip>

```jsx-live
  const App = () => (
  <Stack direction="horizontal">
    <Button tone="primary" size="md" variant="solid">
      Save changes
    </Button>
    <Button tone="primary" size="md" variant="solid">
      <Icons.Undo />
      Revert
    </Button>
    <Button tone="primary" size="md" variant="ghost">

>       Calcel and close

Reviewer: ```suggestion
      Cancel and close
```

    </Button>
  </Stack>
)
```

<Tip tipColor="#46A557" tipBackgroundColor="#F4FAF5" icon={<Icons.Lightbulb />}>
  <div>
    <strong>Do</strong>
    <ul style={{ listStyle: "disc", marginLeft: "20px" }}>
      <li>
        Place the filled primary button to the right. This should represent the
        action most likely taken by a user.
      </li>
      <li>Outlined secondary button to the left of the primary action.</li>
      <li>Icon is on the left of the button.</li>
    </ul>
  </div>
</Tip>

```jsx-live
  const App = () => (
  <Stack direction="horizontal">
    <Button tone="primary" size="md" variant="outline">
      <Icons.Undo />
      Revert
    </Button>
    <Button tone="primary" size="md" variant="solid">
      Save
    </Button>
  </Stack>
)
```

<Tip tipColor="#E5484D" tipBackgroundColor="#FFF5F5" icon={<Icons.Error />}>
  <div>
    <strong>Don't</strong>
    <ul style={{ listStyle: "disc", marginLeft: "20px" }}>
      <li>

>         {" "}
>         Do not place the filled primary button to the left of another
>         option.{" "}

Reviewer: ```suggestion
        Do not place the filled primary button to the left of another
        option.
```

      </li>
      <li>Do not have two primary filled buttons near each other. </li>
      <li>Do not include punctuation in buttons </li>
    </ul>
  </div>
</Tip>

```jsx-live
  const App = () => (
  <Stack direction="horizontal">
    <Button tone="primary" size="md" variant="solid">
      Cancel!
    </Button>
    <Button tone="primary" size="md" variant="solid">

>       Save

Reviewer: ```suggestion
      Save.
```

    </Button>
    <Button tone="primary" as="a" size="sm" variant="link">
      External link
      <Icons.Launch />
    </Button>
  </Stack>
)
```

### Button links

These actions are likely seen when an action is optional to have more or less
information shown.

<Tip tipColor="#46A557" tipBackgroundColor="#F4FAF5" icon={<Icons.Lightbulb />}>
  <div>
    <strong>Do</strong>
    <ul style={{ listStyle: "disc", marginLeft: "20px" }}>
      <li>
        Expand actions should be placed at the lower center or left hand side of
        the element that will expand/collapse.
      </li>
      <li>Link button is using the correct size, 2x-small. </li>
    </ul>
  </div>
</Tip>

```jsx-live
 const App = () => (
  <>
   <div style={{
     width: "300px",

>      height: "60px",

Reviewer: ```suggestion
     height: "40px",
```

     backgroundColor: "#f0f8ff",
     border: "1px dashed #60a5fa",
     borderRadius: "12px",
     display: "flex",
     justifyContent: "center",
     alignItems: "center",
     position: "relative",
   }}>
     <span style={{
       fontSize: "10px",
       color: "#3b82f6",
       letterSpacing: "1px",
     }}>Trigger</span>
  </div>
    <Button tone="primary" as="a" size="2xs" variant="link" marginLeft="220px">
      <Icons.KeyboardArrowDown />
      expand
    </Button>
</>
)
```

<Tip tipColor="#E5484D" tipBackgroundColor="#FFF5F5" icon={<Icons.Error />}>
  <div>
    <strong>Do</strong>
    <ul style={{ listStyle: "disc", marginLeft: "20px" }}>
      <li>The button is too large for this placement</li>
      <li>
        Expansion icons should be placed to the left side of the label to allow

>         for easier scanning by a user.{" "}

Reviewer: ```suggestion
        for easier scanning by a user.
```

      </li>
    </ul>
  </div>
</Tip>

```jsx-live
  const App = () => (
  <>
   <div style={{
     width: "300px",

>      height: "60px",

Reviewer: ```suggestion
     height: "40px",
```

     backgroundColor: "#f0f8ff",
     border: "1px dashed #60a5fa",
     borderRadius: "12px",
     display: "flex",
     justifyContent: "center",
     alignItems: "center",
     position: "relative",
   }}>
     <span style={{
       fontSize: "10px",
       color: "#3b82f6",
       letterSpacing: "1px",
     }}>Trigger</span>
  </div>
    <Button tone="primary" as="a" size="md" variant="link" marginLeft="200px">
      expand
      <Icons.KeyboardArrowDown />
    </Button>
</>
)
```

### Specs

<PropTable id="Button" />

### Accessibility

> The Button component uses onPress\* props to handle all user interactions—mouse

Reviewer: ```suggestion
The Button component uses `onPress*` props to handle all user interactions—mouse
```

> clicks, touch taps, and keyboard presses—consistently. These props (onPress,
> onPressStart, onPressEnd, etc.) ensure a smooth and accessible experience

Reviewer: ```suggestion
clicks, touch taps, and keyboard presses—consistently. These props (`onPress`,
`onPressStart`, `onPressEnd`, etc.) ensure a smooth and accessible experience
```

regardless of the user's input method or device.

```jsx-live
const App = () => (
    <Button tone="primary" size="md" variant="solid" onPress={() => alert("clicked!")}>
      Click me... with your keyboard
    </Button>
)
```

### Accessibility Standards

> #### 1. Perceivable:
> 
> - **Sufficient Color Contrast:**
> - **Text on Button:** The contrast ratio between the button text and the button
>   background should be at least 4.5:1 for normal text and 3:1 for large text.
> - **Button to Background:** The contrast ratio between the button's background
>   color and the surrounding page background should be at least 3:1.
> - **Visual Cues:** Don't rely solely on color to convey information. Use
>   additional visual cues like shapes, patterns, or text labels to ensure the
>   button's meaning is clear to everyone, including people with color blindness.
> - **Focus Indicator:** Provide a clear visual indicator when the button receives
>   keyboard focus. This helps users understand which element is currently active.
> - **WCAG 2.2 has specific criteria for focus indicators:**
> - **2.4.11 Focus Not Obscured (Minimum) (Level AA):** The focus indicator must
>   not be hidden by other content on the page.  
> - **2.4.13 Focus Appearance (Minimum) (Level AA):** The focus indicator must
>   meet certain visibility requirements to be easily seen.    

Reviewer: ```suggestion
**1. Perceivable:**

- **Sufficient Color Contrast:**
  - **Text on Button:** The contrast ratio between the button text and the button
  background should be at least 4.5:1 for normal text and 3:1 for large text.
  - **Button to Background:** The contrast ratio between the button's background
  color and the surrounding page background should be at least 3:1.
- **Visual Cues:** Don't rely solely on color to convey information. Use
  additional visual cues like shapes, patterns, or text labels to ensure the
  button's meaning is clear to everyone, including people with color blindness.
- **Focus Indicator:** Provide a clear visual indicator when the button receives
  keyboard focus. This helps users understand which element is currently active.
  - **WCAG 2.2 has specific criteria for focus indicators:**
    - **2.4.11 Focus Not Obscured (Minimum) (Level AA):** The focus indicator must
  not be hidden by other content on the page.  
    - **2.4.13 Focus Appearance (Minimum) (Level AA):** The focus indicator must
  meet certain visibility requirements to be easily seen.    
```

Author: When you reload your window and format, all indentations are leveled back because white spaces are taken out. Also the `#` notation gives better html hierarchy than the `*` but we can discuss this if this change since we are split between both.

Reviewer: I'm fine with leaving the headers, but the indentation should be fixed if at all possible.

> #### 2. Operable:

Reviewer: ```suggestion
**2. Operable:**
```

- **Keyboard Accessibility:** Buttons must be operable using a keyboard. Users
  should be able to navigate to the button using the Tab key and activate it
  using the Enter or Space key.  
- **Target Size:** Ensure that the button is large enough to be easily
  clickable, especially on touchscreens. WCAG recommends a minimum target size
  of 24x24 CSS pixels.
- **Dragging Movements (2.5.7 Level AA):** If a button's function relies on
  dragging, an alternative method should be provided that doesn't require
  dragging.

> #### 3. Understandable:

Reviewer: ```suggestion
**3. Understandable:**
```

- **Clear Labels:** The button's text label should clearly describe its
  function.
- **Consistent Design:** Use buttons consistently throughout your website. This
  helps users learn the meaning of each button and makes it easier for them to
  navigate your site.
- **Findable Help (3.2.6 Level A):** If users might need help understanding how
  to use a button or what it does, provide clear and easily findable help
  resources.

> #### 4. Robust:

Reviewer: ```suggestion
**4. Robust:**
```

| **Cancel**                              | “Cancel” stops an action without saving pending changes. Never make Cancel red, it is not a destructive action. Cancel should be outline, ghost or link button.                                                                                                                                                        |
| **Delete or Delete (X nouns)**          | Deletes a data so a user can no longer retrieve it. This is different from Remove.                                                                                                                                                                                                                                     |
| **Remove**                              | Removes a relationship, but does not permanently delete data.                                                                                                                                                                                                                                                          |

> | **Expand/Collapse Show more/Show less** | Expands or collapses further details and information. No capitalization is needed for these placements.                                                                                                                                                                                                                |

Reviewer: There should be a line break in between Collapse and Show more

visually as a single unit. They serve to present users with a set of related
choices or actions in a compact and organized way.

> [!TIP]\
> **Do**
>
> This is the correct hierarchy of button groupings. From right to left,

> > Primary, Outlined, and Link button.

Reviewer: Should be in a list

Author: Yes, I noticed if it is just an item, it does not display the list when it is just one item in the list except we add it this way which is clearly wrong

<img width="773" alt="image" src="https://github.com/user-attachments/assets/325f27c9-1dd0-4df2-aafe-6b806f981e22" />

Vs

<img width="775" alt="image" src="https://github.com/user-attachments/assets/76bc40e8-957c-476b-b72f-84fe7c29c105" />

Vs

<img width="780" alt="image" src="https://github.com/user-attachments/assets/d14a33aa-0a81-4269-ac5b-e50d92823d3e" />

Reviewer: Hahahahaha great, leave it as is!
